packages/back-end/src/services/features.ts

```diff
@@ -979,6 +980,114 @@ export function evaluateFeature({
   return results;
 }
 
+export async function evaluateAllFeatures({
+  features,
+  context,
+  attributeValues,
+  environments,
+  groupMap,
+}: {
+  features: FeatureInterface[];
+  context: ReqContext | ApiReqContext;
+  attributeValues: ArchetypeAttributeValues;
+  groupMap: GroupMap;
+  environments?: (Environment | undefined)[];
+}) {
+  const results: { [key: string]: FeatureTestResult }[] = [];
+  const savedGroups = getSavedGroupsValuesFromGroupMap(groupMap);
+
+  const allFeaturesRaw = await getAllFeatures(context);
+  const allFeatures: Record<string, FeatureDefinitionWithProject> = {};
+  if (allFeaturesRaw.length) {
+    allFeaturesRaw.map((f) => {
+      allFeatures[f.id] = {
+        ...f,
+        project: f.project,
+      };
+    });
+  }
+  // change the NODE ENV so that we can get the debug log information:
+  let switchEnv = false;
+  if (process.env.NODE_ENV === "production") {
+    process.env = {
+      ...process.env,
+      NODE_ENV: "development",
+    };
+    switchEnv = true;
+  }
```

Comment:
I added additional checks to make sure it's set back.

Reply:
Ok thats better, but it is still theoretically unsafe if we have other prod checks in the future (from what I can find all the prod checks we currently have are in secrets or other file that probably get loaded when the application boots). More robust ways of doing this would be to:

- change the sdk to have an envOverride, or debugInProd property
- use `worker_threads` npm package. I'll DM you how that would work.